tracing: do not set tags when setting stats

yuzefovich opened the pull request:
**colflow: do not set redundant tag on the tracing span**

This is unnecessary given that we're setting the componentID below that
includes the flowID.

Release justification: low-risk update to new functionality.

Release note: None

**tracing: do not set tags when setting stats**

We no longer need to set tags on the tracing span in order to propagate
stats since we now propagate that info as a whole object (at the moment,
twice - as a Structured payload and as DeprecatedStats protobuf; the
latter will be removed after 21.1 branch is cut).

Addresses: #59379.

Release justification: low-risk update to new functionality.

Release note: None

yuzefovich committed:
colflow: do not set redundant tag on the tracing span

This is unnecessary given that we're setting the componentID below that
includes the flowID.

Release justification: low-risk update to new functionality.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61158)
<!-- Reviewable:end -->

yuzefovich committed:
tracing: do not set tags when setting stats

We no longer need to set tags on the tracing span in order to propagate
stats since we now propagate that info as a whole object (at the moment,
twice - as a Structured payload and as DeprecatedStats protobuf; the
latter will be removed after 21.1 branch is cut).

Release justification: low-risk update to new functionality.

Release note: None

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61158#-:-MURLm7YEj4ZTc0cjdJD:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
